server/src/test/java/io/crate/expression/scalar/string/ReverseFunctionTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar.string;
+
+import org.junit.Test;
+
+import io.crate.expression.scalar.ScalarTestCase;
+
+public class ReverseFunctionTest extends ScalarTestCase {
+    @Test
+    public void test_common_argument() {
+        assertEvaluate("reverse('abcde')", "edcba");
+    }
+
+    @Test
+    public void test_argument_is_null() {
+        assertEvaluateNull("reverse(null)");
+    }
+
+    @Test
+    public void test_argument_is_empty() {
+        assertEvaluate("reverse('')", "");
+    }
+
+    @Test
+    public void test_argument_is_number() {
+        assertEvaluate("reverse(12345)", "54321");
+    }
+
+    @Test
+    public void test_argument_is_other_type() {
+        /* This is not valid in Postgres */
+        assertEvaluate("reverse(12345::numeric)", "54321");
+    }
```

Comment:
```suggestion
```

These are just testing the CrateDB cast semantics, and are not really testing any aspect of the reverse function. The actual argument passed to the function will be a string in both cases.